app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -216,5 +217,44 @@ def check_disconnected
             "is prohibited. If you would like to change this, please update the content setting 'Disconnected mode'."
       fail HttpErrors::BadRequest, _(msg) if Setting[:content_disconnected]
     end
+
+    def filtered_associations
+      nil
+    end
+
+    def filter_association_ids
+      if filtered_associations
+        find_param_arrays.each do |key_path|
+          if (model_class = filtered_associations.with_indifferent_access.dig(*key_path))
+            filtered_ids = model_class.readable.where(:id => params.dig(*key_path)).pluck(:id)
```

Comment:
I made this silently ignore items that can't be read, but now i'm wondering if thats what we should do.  I imagine a UI or cli that would be taking a list of associated Ids and adding one item to it.  With the current logic, if the user can't read one of those items it is silently removed from the content view because they couldn't read it.  Does that make the most sense? or does rejecting the entire request if any items in the association list are unreadable make the most sense?

Reply:
I think we should definitely reject the whole request